In [56]:
# @title Расчёт транзисторного каскада с ОЭ
import math
U_ip = 10  # Питающее напряжение, В
P_max = 350
f = 1000 # частота каскада Гц
I_max = 150  # Максимальный ток коллектора без сигнала, мА
h21 = 100  # Коэффициент усиления транзистора
U_be = 0.66  # Напряжение база-эмиттер, В

- Марка транзистора = КТ503E
- частота работы каскада $f = 1000 Гц$
- Структура транзистора = Кремний n-p-n
- Питающее напряжение $U_{и.п.} = 10В$
- $P_{max} = 350(мВт)$
- $I_{max} = 150 (мА)$
- $h_{21} = 100$
- Сопротивление коллектора $R_{к} = 10\times R_э$
- Напряжение база-эмиттер $U_{бэ} = 0,66В$

In [38]:
# @title 1. Определим максимальную статическую мощность
P_pac_max = 0.8 * P_max
P_pac_max

280.0

$P_{\text{pac.max}} = 0.8 \times P_{\text{max}} = 0.8 \times 350 \, \text{мВт} = 280 \, \text{мВт}$

In [39]:
# @title 2. Определим ток коллектора в статическом режиме (без сигнала):
I_k0 = P_pac_max / (U_ip / 2)
I_k0

56.0

$I_{\text{к0}} = \frac{280 мВт}{\frac{10 В}{2}} \approx 56 \, \text{мА}$

In [40]:
# @title 3. Учитывая, что на транзисторе в статическом режиме (без сигнала) падает половина напряжения питания, вторая половина напряжения питания будет падать на резисторах:
R_ke = (U_ip / 2)/(I_k0/1000)
R_k=80
R_e = 9
R_ke

89.28571428571428

$(R_к+R_э)=\frac{(\frac{U_{и.п.}}{2})}{I_{к0}} = (10/2)/56mA = 5B/0,056A \approx 89 \Omega$

$R_к = 80\Omega$

$R_э = 9\Omega$

In [41]:
# @title 4. Найдем напряжение на коллекторе транзистора без сигнала
U_k0 = (U_ip - I_k0/1000 * R_k)
U_k0

5.52

$U_{к0}=(U_{кэ0}+I_{к0}\times R_э) = (U_{и.п.}-I_{к0}\times R_к) = 10В-0,056А\cdot 80\Omega \approx 5.52В$

In [42]:
# @title 5. Определим ток базы управления транзистором:
I_b = (U_ip/(R_k+R_e))/h21*1000
I_b

1.1235955056179774

$I_б = \frac{I_к}{h_{21}}=\frac{[\frac{U_{и.п.}}{R_к+R_э}]}{h_{21}}=[10/89]/100 \approx 1.123мA$

In [43]:
# @title 6. Полный базовый ток определяется напряжением смещения на базе, которое задается делителем напряжения Rб1,Rб2. Выбираем ток делителя в 10 раз большим тока управления базы:
I_del = 10*I_b
I_del

11.235955056179774

$R_{б1},R{б2}:I_{дел.}=10*I_б \approx 11.23мА$

In [44]:
# @title Тогда полное сопротивление резисторов
R_b1_b2 = U_ip/(I_del/1000)
R_b1_b2

890.0

$R_{б1}+R{б2}=\frac{U_{и.п.}}{I_{дел.}}=10/0.01123A \approx 890 \Omega$

In [48]:
# @title 7. Найдѐм напряжение на эмиттере в режиме покоя
U_e = I_k0/1000 * R_e
U_e

0.504

$U_э=I_{к0}\times R_э = 0,056A * 9\Omega \approx 0,504В$

In [50]:
# @title 8. Определяем напряжение на базе
U_b = U_e + U_be
U_b

1.1640000000000001

$U_б = U_э + U_{бэ} = 0,504B+0,66B=1,164B$

In [54]:
# @title Отсюда, через формулу делителя напряжения находим:
R_b2 = R_b1_b2*U_b/U_ip
R_b1 = R_b1_b2-R_b2
R_b1, R_b2

(786.404, 103.596)

$R_{б2}=(R_{б1}+R{б2})\times \frac{U_б}{U_{и.п.}} = 890\Omega * 1,164B/10B \approx 103.596 \Omega$

$R_{б1}=(R_{б1}+R{б2})-R_{б2} = 890\Omega - 103,596\Omega \approx 789.404\Omega$

В соответсвии с резисторным рядом округляем до ближайшего значения показания резисторов:
- $R_{б2} = 100 \Omega$
- $R_{б1} = 750 \Omega$


In [61]:
# @title 9. Разделительные конденсаторы
X_sr = 0.1 * R_b2
X_sr
X=(1/X_sr)/(2*math.pi*f)*1000000
X

15.363039411936306

$X_{cp} = 0.1 \times R_{б2} = 10\Omega$

$X_{ср} = \frac{1}{2\pi f C_p} = 10\Omega$

$X = \frac{\frac{1}{10}}{6280}\approx 15 \mu$